### PR TITLE
Model, Fixed fineOne incorrect initial order

### PR DESCRIPTION
**Pull request by Yahasana**
initial model data should go ahead `initialize`

### CODE REVIEW

**Comment by mruz** [comment]
But then we will get the `Not found` exception if no result, previously it was `false`.
```php
$u = Users::findOne(['username' => 'abcd']);
```

**Comment by Yahasana** [comment]
Model should be fine now. it's there exact release date for v1.4.0 ? @mruz

**Comment by mruz** [comment]
I'll add some tests for the model and merge after. Just to be sure it's working fine. 
There is no strict release date atm.

**Comment by Yahasana** [comment]
Initial flow test case
```php
class TestModel extends Model
{
    public function __construct($filters = null, array $data = [])
    {
        echo 'model contruct begin';
        parent::__construct($filters, $data);
        echo 'model contruct done';
        print_r($this->all());
    }
    public function onContruct()
    {
        echo 'model contructing...';
        print_r($this->all());
        echo 'model set columnName=onContruct';
        $this->set('columnName', 'onContruct'); // real columnName
    }
    public function initialize()
    {
        echo 'model initialize...';
        print_r($this->all());
        echo 'model set columnName=initialize';
        $this->set('columnName', 'initialize'); // real columnName
    }
}
```

**Comment by mruz** [comment]
I pushed few db/model tests. Could you please rebase your branch?
```
git remote add upstream https://github.com/ice/framework.git
git fetch upstream
git rebase upstream/dev
git push origin patch-10
```

**Comment by Yahasana** [comment]
it'd be bug of unitest. after some investigations, you can't write unitest case for `static` things like as `non-static` one

File: ice/mvc/model.zep [hunk 1]
```diff
@@ -147,14 +151,16 @@ abstract class Model extends Arr implements \Serializable
...
     {
-        var result, instances, data;
+        var result, instances, data, model;
 
         let result = this->db->find(this->from, filters, options),
             instances = [];
 
         if result->count() {
             for data in iterator(result) {
-                let instances[] = create_instance_params(get_called_class(), [null, data]);
```
Comment (mruz) [inline ice/mvc/model.zep:157]: As I remember I had some issues with static implementation on zephir and that's why I used
```zep
create_instance_params(get_called_class(), [null, data]);
```

File: ice/mvc/model.zep [hunk 2]
```diff
@@ -178,13 +184,9 @@ abstract class Model extends Arr implements \Serializable
...
     {
-        var result, model, instance;
-
-        let model = get_called_class(),
-            instance = create_instance(model),
-            result = instance->loadOne(filters, options);
-
-        return result;
+        var model;
+        let model = new static(filters, options);
```
Comment (mruz) [inline ice/mvc/model.zep:188]: Options are not the same as data in construct.

Options are passed to the db::find query
```
// eg. find the latest active user
$user = User::findOne(['status' => 1], ['order' => 'id DESC']);
```

Data is used to populate the fields
```
// fills the $user->data attribute
$user = new User(null, ['username' => 'ice', 'status' => 1]);
```

**Review by mruz — CHANGES_REQUESTED** [review]
It would be good to continue in #183

**Comment by Yahasana** [comment]
new [pr](https://github.com/ice/framework/pull/185)

**Pull request closed without merging**